### PR TITLE
Feat/eda raul

### LINKED ISSUE #2: #1 EDA & Visualización
Limpieza y Preparación de Datos

Descripción:
Realizar un EDA completo incluyendo:

Limpieza
Análisis de valores nulos
Histogramas
Boxplots
Matriz de correlación
Análisis de clases
Conclusiones orientadas a clasificación

Rama sugerida:

feature/eda-nombre1

### PR DESCRIPTION
**Pull request by RaulCtm**
<html>
<body>
<html><head></head><body><h2>Descripción</h2>
<p>Implementación del pipeline de encoding, transformación logarítmica y balanceo de clases. Se preparan los datos del dataset limpio para el modelado baseline aplicando transformación log1p, split estratificado 60/20/20, preprocesado con ColumnTransformer y balanceo con SMOTE. El notebook genera los splits listos para que cualquier compañero pueda entrenar modelos directamente.</p>
<hr>
<h2>Issue relacionada</h2>
<p>Closes #2 #15</p>
<hr>
<h2>Problema que soluciona</h2>
<p>El dataset limpio no podía entrar directamente a sklearn — las variables categóricas eran strings, AgeInDays tenía skewness 2.158 que perjudica a modelos lineales, y At Risk representaba solo el 3.7% con ratio 13.2:1. Además no había una forma de compartir los datos procesados con el equipo sin repetir todo el pipeline.</p>
<hr>
<h2>Cambios realizados</h2>
<ul>
<li>Transformación log1p sobre AgeInDays: skewness reducido de 2.158 a ~0.3</li>
<li>Label Encoding del target (OutcomeClass → 0/1/2/3) con orden fijo de clases</li>
<li>Split estratificado 60/20/20: At Risk preservado al ~3.7% en los tres splits</li>
<li>ColumnTransformer con RobustScaler, OneHotEncoder y OrdinalEncoder — fit solo en train</li>
<li>SMOTE sobre X_train_prep: At Risk de 2.462 → 32.588 ejemplos sintéticos</li>
<li>Guardado de splits en <code>splits_procesados.npz</code> para uso compartido del equipo</li>
</ul>
<hr>
<h2>Dependencias o librerías instaladas</h2>

Librería | Versión | Motivo
-- | -- | --
scikit-learn | latest | RobustScaler, OHE, OrdinalEncoder, LabelEncoder, train_test_split
imbalanced-learn | latest | SMOTE para oversampling de At Risk
numpy | latest | Guardado de arrays en .npz

<hr>
<h2>Notas para el revisor</h2>
<ul>
<li>El preprocesador hace <code>fit_transform</code> solo en train — val y test usan solo <code>transform</code></li>
<li>SMOTE se aplica después del preprocesador porque necesita datos numéricos para interpolar</li>
<li>Val y test no se tocan en ningún momento — representan la distribución real del problema</li>
<li><code>X_train_bal</code> tiene 130.352 filas porque SMOTE generó 63.845 ejemplos sintéticos de At Risk</li>
<li>KNN no tiene <code>class_weight</code>, en el baseline se evaluará únicamente con la versión SMOTE</li>
<li>Si <code>splits_procesados.npz</code> pesa más de 100MB añadirlo al <code>.gitignore</code> y compartir por Drive</li>
</ul></body></html>
</body>
</html><html>
<body>
<html><head></head><body><h2>Descripción</h2>
<p>Implementación del pipeline de encoding, transformación logarítmica y balanceo de clases. Se preparan los datos del dataset limpio para el modelado baseline aplicando transformación log1p, split estratificado 60/20/20, preprocesado con ColumnTransformer y balanceo con SMOTE. El notebook genera los splits listos para que cualquier compañero pueda entrenar modelos directamente.</p>
<hr>
<h2>Issue relacionada</h2>
<p>Closes #2 #15</p>
<hr>
<h2>Problema que soluciona</h2>
<p>El dataset limpio no podía entrar directamente a sklearn — las variables categóricas eran strings, AgeInDays tenía skewness 2.158 que perjudica a modelos lineales, y At Risk representaba solo el 3.7% con ratio 13.2:1. Además no había una forma de compartir los datos procesados con el equipo sin repetir todo el pipeline.</p>
<hr>
<h2>Cambios realizados</h2>
<ul>
<li>Transformación log1p sobre AgeInDays: skewness reducido de 2.158 a ~0.3</li>
<li>Label Encoding del target (OutcomeClass → 0/1/2/3) con orden fijo de clases</li>
<li>Split estratificado 60/20/20: At Risk preservado al ~3.7% en los tres splits</li>
<li>ColumnTransformer con RobustScaler, OneHotEncoder y OrdinalEncoder — fit solo en train</li>
<li>SMOTE sobre X_train_prep: At Risk de 2.462 → 32.588 ejemplos sintéticos</li>
<li>Guardado de splits en <code>splits_procesados.npz</code> para uso compartido del equipo</li>
</ul>
<hr>
<h2>Dependencias o librerías instaladas</h2>

Librería | Versión | Motivo
-- | -- | --
scikit-learn | latest | RobustScaler, OHE, OrdinalEncoder, LabelEncoder, train_test_split
imbalanced-learn | latest | SMOTE para oversampling de At Risk
numpy | latest | Guardado de arrays en .npz

<hr>
<h2>Notas para el revisor</h2>
<ul>
<li>El preprocesador hace <code>fit_transform</code> solo en train — val y test usan solo <code>transform</code></li>
<li>SMOTE se aplica después del preprocesador porque necesita datos numéricos para interpolar</li>
<li>Val y test no se tocan en ningún momento — representan la distribución real del problema</li>
<li><code>X_train_bal</code> tiene 130.352 filas porque SMOTE generó 63.845 ejemplos sintéticos de At Risk</li>
<li>KNN no tiene <code>class_weight</code>, en el baseline se evaluará únicamente con la versión SMOTE</li>
<li>Si <code>splits_procesados.npz</code> pesa más de 100MB añadirlo al <code>.gitignore</code> y compartir por Drive</li>
</ul></body></html>
</body>
</html>

### CODE REVIEW

**Review by MichelleGel — APPROVED** [review]
bien, siguiente paso
